src/resolvers.ts

```diff
@@ -39,22 +39,38 @@ export const resolvers = {
       return user;
     },
 
-    users: async (_parent: never, args: { maxUsers?: number }, context: { token: string }) => {
+    users: async (_parent: never, args: { maxUsers?: number; step?: number }, context: { token: string }) => {
       await validateContext(context);
 
-      const max = args.maxUsers ? args.maxUsers : 10;
-      if (args.maxUsers == 0) {
-        return [];
+      const maxUsers = args.maxUsers !== undefined ? args.maxUsers : 3;
+      if (maxUsers <= 0) {
+        throw new CustomError(400, 'Requisição inválida', 'Tamanho da página deve ser maior que zero');
+      }
+
+      const pageSize = maxUsers;
+      const step = args.step ? args.step : 0;
+      const totalUsers = await dataSource.getRepository(User).count();
+
+      if (step >= totalUsers) {
+        throw new CustomError(400, 'Requisição inválida', 'Erro ao acessar usuário');
       }
 
       const users = await dataSource.getRepository(User).find({
-        take: max,
+        skip: step,
+        take: pageSize,
         order: {
           name: 'ASC',
-        }
+        },
       });
 
-      return users;
+      const after = step + pageSize < totalUsers;
```

Comment:
só `after` acho que é pouco descritivo para booleano, aqui uma sugestão de leitura: https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig